pkg/asset/machines/azure/machines.go

```diff
@@ -75,11 +79,67 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 				// we don't need to set Versions, because we control those via operators.
 			},
 		}
-
+		*machineSetProvider = *provider
 		machines = append(machines, machine)
 	}
+	replicas := int32(total)
+	failureDomains := []machinev1.AzureFailureDomain{}
+	sort.Strings(mpool.Zones)
+	for _, zone := range mpool.Zones {
+		domain := machinev1.AzureFailureDomain{
+			Zone: zone,
+		}
 
-	return machines, nil
+		failureDomains = append(failureDomains, domain)
+	}
+	machineSetProvider.Zone = nil
+	machineSetProvider.InternalLoadBalancer = fmt.Sprintf("%s-internal", clusterID)
```

Comment:
We should probably refactor in the future because `ConfigMasters` and this configuration are likely to become out of sync again

Reply:
Fixed it.